app/utils/url.js

```diff
@@ -43,6 +43,26 @@ export function isYoutubeLink(link) {
     return link.trim().match(ytRegex);
 }
 
+export async function getShortenedImageLink(link) {
+    return new Promise(((resolve, reject) => {
+        const xhr = new XMLHttpRequest();
+
+        xhr.onreadystatechange = () => {
+            if (xhr.readyState !== XMLHttpRequest.DONE) {
+                return;
+            }
+            resolve(xhr.responseURL);
+        };
+
+        xhr.onerror = () => {
+            reject('');
+        };
+
+        xhr.open('GET', link);
```

Comment:
Would a `HEAD` request be usable here? No point in loading the whole linked file if it's not an image

Reply:
@hmhealey That's a good point. I'll give it a try.